File: codingame/easy/powerOfThor/powerOfThor.pl
#use strict;
use warnings;
#use diagnostics;
use 5.20.1;

select(STDOUT); $| = 1; # DO NOT REMOVE

my $tokens;
my $x;
my $y;

chomp($tokens=<STDIN>);
my ($light_x, $light_y, $initial_tx, $initial_ty) = split(/ /,$tokens);

while (1) {
    chomp(my $remaining_turns = <STDIN>); # The remaining amount of turns Thor can move. Do not remove this line.
    $y = "";
    $x = "";
    if ($initial_ty < $light_y) {
        $y = "S";
        $initial_ty++;
        print STDERR "$y, $initial_ty\n";
        }
    elsif ($initial_ty > $light_y) {
        $y = "N";
        $initial_ty--;
        }
    else {
        $y = "";
    }
    
    if ($initial_tx < $light_x) {
        $x = "E";
        $initial_tx--;
        }
    elsif ($initial_tx > $light_x) {
        $x = "W";
        $initial_tx++;
        }
    else {
        $x = "";
    }
    print STDERR "$x, $y\n";

    #$initial_ty = ($initial_ty < $light_y) ? $initial_ty++ : $initial_ty;
    #$$initial_ty = ($initial_ty > $light_y) ? $initial_ty-- : $initial_ty;
    #$initial_tx = ($initial_tx < $light_x) ? $initial_tx++ : $initial_tx;
    #print STDERR "$initial_tx - $light_x\n";
    #$initial_tx = ($initial_tx > $light_x) ? $initial_tx-- : $initial_tx;
    #print STDERR "$initial_tx\n";
    print "$y$x\n";
}

File: codingame/easy/powerOfThor/powerOfThor.go
package main
import "fmt"

func main() {

    var lightX, lightY, initialTX, initialTY int
    fmt.Scan(&lightX, &lightY, &initialTX, &initialTY)
    var ns, ew string
    
    for {

        var remainingTurns int
        fmt.Scan(&remainingTurns)
        
        if initialTX < lightX {
            ew = "E"
            initialTX++
        } else if initialTX > lightX {
            ew = "W"
            initialTX--
        } else {
            ew = ""
        }
        
        if initialTY < lightY {
            ns = "S"
            initialTY++
        } else if initialTY > lightY {
            ns = "N"
            initialTY--
        } else {
            ns = ""
        }
        
        fmt.Println(ns + ew)
    }
}